libextract/coretools.py

```diff
@@ -20,7 +20,12 @@ def proc(pairs):
     return proc
 
 
-def pipeline(data, functions):
-    for item in functions:
-        data = item(data)
-    return data
+def pipeline(data, *args):
+    try:
+        for arg in args[0:]:
+            data = arg(data)
+    except(TypeError):
```

Comment:
I see, but IMHO I strongly suggest against function signatures like that, as it, IMO is less readable in the second form. Also, the first one encourages code as data philosophy.